File: LeetCode-Programs/N-th Tribonacci Number.cpp
class Solution {
public:
    int tribonacci(int n) {
        //tle
        /*if (n==0)
            return  0;
        if (n==1 || n==2)
            return 1;            
        return tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3);*/
      
        int arr[n+3];
        arr[0]=0;
        arr[1]=1;
        arr[2]=1;
        for(int i=3;i<=n;i++)
            arr[i]=arr[i-1]+arr[i-2]+arr[i-3];
        return arr[n];
    }
};

File: LeetCode-Programs/N-th Tribonacci Number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        '''if n==0:
            return  0
        if n==1 or n==2:
            return 1            
        return self.tribonacci(n-1) + self.tribonacci(n-2) + self.tribonacci(n-3)'''
        new_list=[0,1,1]
        for i in range(3, n+1):
            new_list.append(new_list[i-3]+new_list[i-2]+new_list[i-1])
        return new_list[n]
            
